src/core/sock/sockinfo_tcp.cpp

```diff
@@ -428,6 +428,13 @@ void sockinfo_tcp::set_xlio_socket(const struct xlio_socket_attr *attr)
 void sockinfo_tcp::set_entity_context(entity_context *ctx)
 {
     m_entity_context = ctx;
+    // To reuse Ultra API path of TX completions
+    m_p_group = ctx;
+
+    assert(m_ops == m_ops_tcp);
+    m_ops_tcp = new sockinfo_tcp_ops_thread(this);
```

Comment:
should we move here rx path and maybe other stuff as well?

Reply:
We can consider this. Currently RX path is done via lwip callback. With full access to sockinfo_tcp. Moving this logic to ops::recv() can limit us. Anyway, ops functionality can be extended if needed.